pkg/network/tracer/tracer.go

```diff
@@ -121,8 +121,10 @@ func NewTracer(config *config.Config, telemetryComponent telemetryComponent.Comp
 // newTracer is an internal function used by tests primarily
 // (and NewTracer above)
 func newTracer(cfg *config.Config, telemetryComponent telemetryComponent.Component) (_ *Tracer, reterr error) {
-	if _, err := tracefs.Root(); err != nil {
-		return nil, fmt.Errorf("system-probe unsupported: %s", err)
+	if !cfg.EnableEbpfless {
+		if _, err := tracefs.Root(); err != nil {
+			return nil, fmt.Errorf("system-probe unsupported: %s", err)
+		}
 	}
```

Comment:
The current tracer is complicated enough, i think this change (repeated a couple of times in the function, makes the function even more complicated
I do recommend changing it, or splitting to ebpless and the rest

Reply:
Going to move this check to `newEbpfTracer`